eng/Versions.props

```diff
@@ -41,15 +41,15 @@
   <PropertyGroup>
     <!-- Code analysis dependencies -->
     <MicrosoftCodeAnalysisAnalyzersVersion>3.3.3</MicrosoftCodeAnalysisAnalyzersVersion>
-    <MicrosoftCodeAnalysisCSharpCodeStyleVersion>4.3.0-1.22206.2</MicrosoftCodeAnalysisCSharpCodeStyleVersion>
-    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>4.3.0-1.22206.2</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
-    <MicrosoftCodeAnalysisCSharpVersion>4.3.0-1.22206.2</MicrosoftCodeAnalysisCSharpVersion>
+    <MicrosoftCodeAnalysisCSharpCodeStyleVersion>4.4.0-1.22356.23</MicrosoftCodeAnalysisCSharpCodeStyleVersion>
+    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>4.4.0-1.22356.23</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
+    <MicrosoftCodeAnalysisCSharpVersion>4.4.0-1.22356.23</MicrosoftCodeAnalysisCSharpVersion>
     <MicrosoftCodeAnalysisNetAnalyzersVersion>7.0.0-preview1.22329.1</MicrosoftCodeAnalysisNetAnalyzersVersion>
-    <MicrosoftCodeAnalysisVersion>4.3.0-1.22206.2</MicrosoftCodeAnalysisVersion>
+    <MicrosoftCodeAnalysisVersion>4.4.0-1.22356.23</MicrosoftCodeAnalysisVersion>
```

Comment:
@jkoritzinsky, @ViktorHofer, a recent 4.4.x version of Microsoft.CodeAnalysis is needed for this (in particular for pretty much all of the tests to be successful), and we need this to go in for .NET 7.  When can we update these versions?

Reply:
If we want, we can build the tests against the new Roslyn version. The generator needs to build against an older Roslyn version until Arcade updates to an SDK with the Roslyn version and there’s a VS version with the Roslyn build, so it’s a ways out for a 4.4.x Roslyn build.

If we don’t care about the generator working in VS (or if this is higher priority than that), then we can push other repos to consume a new enough Roslyn so we won’t break upstack repos.

Reply:
> If we want, we can build the tests against the new Roslyn version. 

Thanks.  Do we have any precedent for that?

Reply:
We don't have precedent yet, but I think it's honestly the best solution and I've pushed for it in a variety of cases.